### PR TITLE
Improve error handling for non basic authentication

### PR DESCRIPTION
**Pull request by mwrock**
The current authentication failure messaging is specific to basic authentication and provides advise for turning off encryption on the target node. Failures for Negotiate authentication are not caught and just surface the `WinRMAuthorizationError`. This PR specifically catches `WinRMAuthorizationError` and provides a different Hint to the user when not using Basic auth.

### CODE REVIEW

**Comment by mwrock** [comment]
Odd, the tests pass on my box and i am using `bundle exec`. Will investigate...may be a windows thing.

File: Rakefile [hunk 1]
```diff
@@ -4,11 +4,16 @@ Bundler::GemHelper.install_tasks
 begin
   require 'rspec/core/rake_task'
 
-  task :default => :spec
+  task :default => [:unit_spec, :functional_spec]
 
   desc "Run all specs in spec directory"
```
Comment (ksubrama) [inline Rakefile]: Fix descriptions

File: spec/unit/knife/winrm_spec.rb [hunk 2]
```diff
@@ -272,9 +272,10 @@
           expect(exit_code).to be_zero
         end
 
-        it "exits with 100 if command execution raises an exception other than 401" do
+        it "exits with 100 and nom hint if command execution raises an exception other than 401" do
```
Comment (ksubrama) [inline spec/unit/knife/winrm_spec.rb]: typo "no hint"

**Comment by ksubrama** [comment]
Check appveyor.yml  It's doing bundle exec rake spec  travis.yml doesn't specify anything.  I think your rake file cleanup may be causing them to run different tests (maybe?).

**Comment by mwrock** [comment]
Heh. Just took a closer look at the Travis output. It's because the build runs on Linux and I ran the tests in windows. Just need to change the test to fake the windows platform.

**Comment by ksubrama** [comment]
D'oh - I thought the unit test helpers already did that in knife-windows.  I guess not.

**Comment by mwrock** [comment]
squashed things up and working to fix new build breaks as a result of `tags` being added to core bootstrap. Will push that fix ini a separate commit.

**Comment by smurawski** [comment]
:+1:

File: Rakefile [hunk 3]
```diff
@@ -4,11 +4,16 @@ Bundler::GemHelper.install_tasks
 begin
   require 'rspec/core/rake_task'
 
-  task :default => :spec
+  task :default => [:unit_spec, :functional_spec]
```
Comment (mcquin) [inline Rakefile]: I'm for this change, but against making it in this PR. It's not within scope.

Comment (mwrock) [inline Rakefile]: thats cool. I kept it in a separate commit and can merge separately

Comment (chefsalim) [inline Rakefile]: Is it basically a no-op or does it actually make a difference?

Comment (mwrock) [inline Rakefile]: @chefsalim not sure i get the question, but it just separates functional and unit tests into separate tasks but the default task (what ci uses) will still run both.

Comment (smurawski) [inline Rakefile]: @chefsalim it doesn't break anything but improves the dev experience.

File: lib/chef/knife/bootstrap_windows_base.rb [hunk 4]
```diff
@@ -170,6 +170,12 @@ def self.included(includer)
               Chef::Config[:knife][:bootstrap_vault_item][vault].push(item)
               Chef::Config[:knife][:bootstrap_vault_item]
             }
+
+          option :tags,
+            :long => "--tags TAGS",
+            :description => "Comma separated list of tags to apply to the node",
+            :proc => lambda { |o| o.split(/[\s,]+/) },
+            :default => []
```
Comment (mcquin) [inline lib/chef/knife/bootstrap_windows_base.rb]: Why was this added, @mwrock ?

Comment (mcquin) [inline lib/chef/knife/bootstrap_windows_base.rb]: I see now, it's for the `first_boot` change. I think this should go in a separate PR, too.

Comment (mwrock) [inline lib/chef/knife/bootstrap_windows_base.rb]: yup. this is gonna be 3 PRs. Apologies for not doing that up front and thanks for noticing!

Comment (mwrock) [inline lib/chef/knife/bootstrap_windows_base.rb]: It does feel like this could be "dried up" Seems like we basically have to inherit via copy/paste when new properties are added to https://github.com/chef/chef/blob/master/lib/chef/knife/bootstrap.rb

would be nice if we could add some DSL like:
```
include_option :runlist
include_option :tags
```

That would also mean not having to do the crazy versioning filters in the tests too

File: lib/chef/knife/winrm.rb [hunk 5]
```diff
@@ -29,6 +29,13 @@ class Winrm < Knife
 
       include Chef::Knife::WinrmCommandSharedFunctions     
 
+      FAILED_BASIC_HINT ||= "Hint: Please check winrm configuration 'winrm get winrm/config/service' AllowUnencrypted flag on remote server."
```
Comment (chefsalim) [inline lib/chef/knife/winrm.rb]: Reduce the line length?  It is getting truncated in the browser view.

File: lib/chef/knife/winrm.rb [hunk 6]
```diff
@@ -29,6 +29,13 @@ class Winrm < Knife
...
+      FAILED_NOT_BASIC_HINT ||= <<-eos.gsub /^\s+/, ""
```
Comment (chefsalim) [inline lib/chef/knife/winrm.rb]: Left justify the text so we don't have to do a gsub?

Comment (mwrock) [inline lib/chef/knife/winrm.rb]: I personally prefer the `gsub` (though I'm not crazy about it) over losing the indentation.

Any one else have opinions here?

Comment (mcquin) [inline lib/chef/knife/winrm.rb]: I'm fine with the `gsub` as well. There are occurrences of both `gsub`bing and left-justifying all over our code bases.

**Comment by mcquin** [comment]
LGTM

File: lib/chef/knife/winrm.rb (hunk 6 above)
Comment (chefsalim) [inline lib/chef/knife/winrm.rb]: Not a big deal - I'm fine with the aestheticly pleasing way :)

Comment (mwrock) [inline lib/chef/knife/winrm.rb]: cool
